tests/test_class.py

```diff
@@ -25,9 +25,18 @@ def test_instance(msg):
 
 
 def test_type():
-    assert m.check_type(True) == m.DerivedClass1
-    with pytest.raises(RuntimeError):
-        m.check_type(False)
+    assert m.check_type(1) == m.DerivedClass1
+    with pytest.raises(RuntimeError) as execinfo:
+        m.check_type(0)
+
+    assert ('pybind11::detail::get_type_info: unable to find type info '
+            'for "test_submodule_class_(module&)::Invalid"' in str(execinfo.value))
+
+    with pytest.raises(RuntimeError) as execinfo:
+        assert m.check_type(2) == int
+
+    assert ('pybind11::detail::get_type_info: unable '
```

Comment:
Wait, this is a runtime error, and not a compilation error? We can figure out at compile time that this is going to fail, right?

Reply:
I'm only addressing one point at a time - currently, yes. In the future, probably no. ;)

Reply:
(Sorry, just got an email with changes and saw you did effort to test for runtime errors. Too impatient, then ;-) )